modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCFeatureSource.java

```diff
@@ -538,12 +538,12 @@ protected ReferencedEnvelope getBoundsInternal(Query query) throws IOException {
     }
 
     @Override
-    protected boolean canFilter() {
+    protected boolean canFilter(Query query) {
```

Comment:
Not quite what I had in mind, the API change should be additive:
```
protected boolean canFilter() { return true; }
protected boolean canFilter(Query query) { return canFilter(); }
```

Reply:
Agreed, reworked.